src/test/java/net/openhft/chronicle/queue/TestDeleteQueueFile.java

```diff
@@ -192,38 +198,279 @@ public void firstAndLastIndexAreRefreshedAfterForceRefreshInterval() throws IOEx
 
             // delete the first store
             Files.delete(Paths.get(firstCycle.filename));
 
             // using old cached value
-            assertEquals(Long.toHexString(firstCycle.firstIndex), Long.toHexString(queue.firstIndex()));
-            assertEquals(Long.toHexString(firstCycle.firstIndex), Long.toHexString(queue.firstIndex()));
+            assertEquals(toHexString(firstCycle.firstIndex), toHexString(queue.firstIndex()));
+            assertEquals(toHexString(firstCycle.firstIndex), toHexString(queue.firstIndex()));
 
             // wait for cache to expire
             Jvm.pause(260);
 
             // using correct value
-            assertEquals(Long.toHexString(secondCycle.firstIndex), Long.toHexString(queue.firstIndex()));
+            assertEquals(toHexString(secondCycle.firstIndex), toHexString(queue.firstIndex()));
         }
     }
 
     @Test
     public void tailingThroughDeletedCyclesWillRefreshThenRetry_Writable() throws IOException {
+        expectException("The current cycle seems to have been deleted from under the queue, scanning to find the next remaining cycle");
         tailingThroughDeletedCyclesWillRefreshThenRetry(qwcd -> qwcd.queue);
     }
 
-    @Ignore("we can no longer recover from the case where someone has deleted a file from underneath us, and not called refreshDirectoryListing, and we are readOnly")
     @Test
     public void tailingThroughDeletedCyclesWillRefreshThenRetry_ReadOnly() throws IOException {
+        expectException("The current cycle seems to have been deleted from under the queue, scanning to find the next remaining cycle");
         tailingThroughDeletedCyclesWillRefreshThenRetry(qwcd -> SingleChronicleQueueBuilder.binary(qwcd.queue.fileAbsolutePath())
                 .rollCycle(RollCycles.FAST_DAILY)
                 .readOnly(true)
                 .build());
     }
 
+    @Test
+    public void deletingOldFilesChaosTest() throws InterruptedException {
+        ignoreException("The current cycle seems to have been deleted from under the queue, scanning to find the next remaining cycle");
+        final int numberOfCycles = 300;
+        final AtomicBoolean running = new AtomicBoolean(true);
+        try (QueueWithCycleDetails queueWithCycleDetails = createQueueWithNRollCycles(numberOfCycles, null)) {
+            Thread backwardTailerThread = new Thread(() -> new QueueTailer(running, queueWithCycleDetails, TailerDirection.BACKWARD));
+            Thread forwardTailerThread = new Thread(() -> new QueueTailer(running, queueWithCycleDetails, TailerDirection.FORWARD));
+            Thread deleterThread = new Thread(() -> progressivelyTruncateOldRollCycles(queueWithCycleDetails));
+
+            backwardTailerThread.start();
+            forwardTailerThread.start();
+            deleterThread.start();
+            deleterThread.join();
+            running.set(false);
+            forwardTailerThread.join();
+            backwardTailerThread.join();
+        }
+    }
```

Comment:
This test progressively deletes cycles in batches from the beginning of the queue. It probably mirrors the way sensible people archive old roll cycles.

While that's occurring, we have two threads continually scanning over the queue in both directions and validation that they saw all the remaining roll cycles. (this is a bit hard to check so we ensure they saw all the roll cycles not deleted when they reached the start/end of the queue.